### DISCUSSION TITLE
Group only in the presence of a specific package update

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#18018] [Request Help] [answered]
In my Spring Boot projects, I want to group the updates to SB, its dependency management plugin, and three of our internal libraries that need to be updated for it. To do so, we have this package rule:
```
    {
      "matchPackagePatterns": [
        "^org\\.springframework\\.boot(:.*|$)",
        "^io\\.spring\\.dependency-management(:.*|$)",
        "^tld\\.domain\\.sub[.:].*$"
      ],
      "matchDatasources": [
        "maven"
      ],
      "groupName": "Spring (Boot) and team-internal dependencies following Spring Boot",
      "groupSlug": "spring-boot-related"
    },
```

Now, I have an update for our internal library, unrelated to Spring Boot. Is there a way to only group if a particular package gets updated?

**Comment by rarkins** [comment, marked as answer]
No, but I think you'll find that if your internal library is the only package in the branch then the PR title will reflect that and not the group name. Branch name will remain per the group though.

What do you hope to benefit from "not grouping" in the scenario where your internal library is the only one?
